pilot/pkg/config/coredatamodel/discovery.go

```diff
@@ -38,35 +38,104 @@ var (
 	_ model.ServiceDiscovery = &MCPDiscovery{}
 )
 
-// DiscoveryOptions stores the configurable attributes of a Control
-type DiscoveryOptions struct {
-	Env          *model.Environment
-	ClusterID    string
-	DomainSuffix string
-}
-
-// mixerEnabled checks to see if mixer is enabled in the environment
-// so we can set the UID on eds endpoints
-func (o *DiscoveryOptions) mixerEnabled() bool {
-	return o.Env != nil && o.Env.Mesh != nil && (o.Env.Mesh.MixerCheckServer != "" || o.Env.Mesh.MixerReportServer != "")
-}
-
 // MCPDiscovery provides discovery interface for SyntheticServiceEntries
 type MCPDiscovery struct {
 	*SyntheticServiceEntryController
-	*DiscoveryOptions
+	cacheMutex sync.RWMutex
+	// keys [namespace][endpointIP]
+	cacheByEndpointIP map[string]map[string][]*model.ServiceInstance
```

Comment:
why need namespace?

Reply:
How do you handle multi cluster scenario, pods in different clusters may have same ip address

Reply:
This discovery only serves a single galley (for every galley we start a separate instance) and galley only serves a single cluster.

Reply:
clusterA   galley A    -- sink A      
                                                  --
                                                         ---- > service registry aggregate   
                                                  --
clusterB    galley B   -- sink B

IC, but still think the namespaces is not necessary since hostname can distinguish k8s services.

Reply:
Assuming all the endpoint IPs are unique within a cluster, I agree the namespaces seems a bit redundant.